generator/sbpg/test_structs.py

```diff
@@ -50,13 +56,29 @@ def filepath(self):
     filepath, filename = "/".join(split[:-1]), "_".join(split[-2:])
     return (filepath, filename)
 
+  def to_json(self):
+    return {
+      'description': self.description,
+      'generated_on': self.generated_on,
+      'package': self.package,
+      'tests': [test.to_json() for test in self.tests],
+    }
+
+  def write(self, filename, exists_ok=False):
+    file_exists = os.path.exists(filename)
+    assert exists_ok or not file_exists
+    data = self.to_json()
+    try:
+      with open(filename, "w") as f:
+        yaml.dump(data, f)
```

Comment:
use yaml formatter settings, or ruamel.yaml, for prettier yaml, or post-process with a yaml reformatter.

Reply:
Spawned https://swift-nav.atlassian.net/browse/INFRA-468